React/game/src/app/components/Screen/Game/index.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+
+import Board from './Board';
+import styles from './styles.scss';
+
+class Game extends Component {
+  state = {
+    moves: 0,
+    gameEnd: false,
+    turn: 'X',
+    turnWinner: '',
+    history: [
+      {
+        board: Array(9).fill(''),
+        id: 0
+      }
+    ]
+  };
+
+  mouseClick = place => {
+    const history = this.state.history.slice(0, this.state.moves + 1);
+    const current = history[history.length - 1];
+    const board = current.board.slice();
+
+    if (board[place] === '' && !this.state.gameEnd) {
+      board[place] = this.state.turn;
+      this.setState({
+        history: history.concat([
+          {
+            board,
+            id: history.length
+          }
+        ]),
+        turn: this.state.turn == 'X' ? 'O' : 'X',
+        moves: ++this.state.moves
+      });
+    }
+
+    const result = this.winner();
+
+    if (result === 'X') {
+      this.setState({
+        gameEnd: true,
+        turnWinner: result
+      });
+    } else if (result === 'O') {
+      this.setState({
+        gameEnd: true,
+        turnWinner: result
+      });
+    }
+  };
+
+  winner = () => {
+    const serialWin = [
+      [0, 1, 2],
```

Comment:
you could extract this in a `constants.js` file in the `Game` folder

Reply:
👍